optimization/physical_constants.py

```diff
@@ -3,6 +3,102 @@
+              'Cp'  : (0.255, 977.66),
+              # temperature limit for curve fits
+              'T_limits' : (900, 1200)
+             }
+    
+    W_e = 40000 # target electrical power [W]
+
+    ###############################################################################
+    #                                                                             #
+    #                     Given Parameters From Power Cycle                       #
+    #                                                                             #
+    ###############################################################################
+    m_dot = 0.75  # coolant flow [kg/s]
+    Q_therm = 131000  # core thermal power [W]
+    eta = W_e / Q_therm
+    T_in = 962.9  # core inlet temp [K] (from power cycle model)
+    T_out = 1100  # core outlet temp [K] (from power cycle model)
+    T = T_in + (T_out - T_in) / 2  # bulk coolant temp. [K]
+    rho = 86.96  # coolant density [kg/m^3]
+    k_cool = 0.086  # coolant conductivity [W/m-k]
+    Cp_cool = 1242  # coolant specific heat [J/kg-k]
+    mu = 0.00004079  # coolant viscosity [kg/m-s]
+    Pr = Cp_cool * mu / k_cool  # coolant Prandtl numbe[-]
+    P_in = 1.79064e7  # inlet pressure [Pa]
+    P_out = 1.74229e7  # outlet pressure [Pa]
+    dp_allowed = P_in - P_out  # pressure drop limit [Pa]
+    default = True
+
+    def __init__(self, T, P, mass_flow, thermal_power):
+        """Inialize FlowProperties class and load required flow property data.
+
+        Modified Attributes:
+        -----------------------
+            m_dot: (float) mass flow rate [kg/s]
+            Q_therm: (float) required thermal reactor power [W]
+            T: (float) bulk coolant temperature [K]
+            P: (float) bulk coolant pressure [Pa]
+            dp_allowed: (float) power-cycle constrained dp [Pa]
+            mass: (float) required reactor mass
+        """
+        self.m_dot = mass_flow
+        self.Q_therm = thermal_power
+        self.T = float(T)
+        self.P = float(P) * 1e3
+        self.secondary_properties()
+        self.dP_allowed = 0
+        t_limit = self.coeffs['T_limits']
+        # if the input temperature is out of range of the fit, print a warning
+        # message
+        if self.T < t_limit[0] or self.T > t_limit[1]:
+            print("Warning T outside of fit range. Consider re-calculating your\
+ fit coeffs. to include this temperature!")
+
+    def secondary_properties(self):
+        """Calculate secondary properties from primary flow properties
+        
+        Modified Attributes:
+        --------------------
+            Cp: (float) specific heat [kJ/kg-K]
+            mu: (float) dynamic viscosity [kg/m-s]
+            k_cool: (float) coolant conductivity [W/m-k]
+            rho: (float) coolant density [kg/m^3]
+            Pr: (float) cooland Prandtl number [-]
+        """
+        # calculate Cp, mu, k, rho, Pr    
+        self.Cp = self.evaluate_fit('Cp')
```

Comment:
This seems a little verbose... could you simply return all of them from a single call to `evaluate_fit`?  Or even setup the coefficients so that its a single vector operation to calculate them at once?

Reply:
I set those defaults before the constructor so I could access the default flow properties without instantiating the class. Is this bad form?